src/services/docGenService.ts

```diff
@@ -174,27 +175,14 @@ export class DocGenService {
     }
   }
 
-  public getProject(): DBTProject | undefined {
-    if (!window.activeTextEditor) {
-      return undefined;
-    }
-    const currentFilePath = window.activeTextEditor.document.uri;
-    return this.dbtProjectContainer.findDBTProject(currentFilePath);
-  }
-
-  public async getDocumentation(
-    eventMap: Map<string, ManifestCacheProjectAddedEvent>,
-  ): Promise<DBTDocumentation | undefined> {
-    if (window.activeTextEditor === undefined || eventMap === undefined) {
+  public async getDocumentation(): Promise<DBTDocumentation | undefined> {
+    if (window.activeTextEditor === undefined) {
       return undefined;
     }
 
     const currentFilePath = window.activeTextEditor.document.uri;
-    const project = this.getProject();
-    if (project === undefined) {
-      return undefined;
-    }
-    const event = eventMap.get(project.projectRoot.fsPath);
+
+    const event = this.dbtProjectService.getEventByCurrentProject();
```

Comment:
Maybe safer to pass down the url of the document to this method.

Reply:
fixed